app/assets/javascripts/admin/controllers/admin-web-hooks-show.js.es6

```diff
@@ -0,0 +1,42 @@
+import { popupAjaxError } from 'discourse/lib/ajax-error';
+import computed from 'ember-addons/ember-computed-decorators';
+
+export default Ember.Controller.extend({
+  needs: ['adminWebHooks'],
+
+  @computed('model.isSaving', 'saved')
+  savingStatus(isSaving, saved) {
+    return isSaving ? I18n.t('saving') : (saved ? I18n.t('saved') : '');
+  },
+
+  @computed('model.isNew')
+  saveButtonText(isNew) {
+    return isNew ? I18n.t('admin.web_hooks.create') : I18n.t('admin.web_hooks.save');
+  },
+
+  actions: {
+    save() {
+      this.set('saved', false);
+
+      const model = this.get('model');
+      return model.save().then(() => {
+        this.set('saved', true);
+        this.get('controllers.adminWebHooks').get('model').addObject(model);
+      }).catch(popupAjaxError);
```

Comment:
Not sure whether it's needed at all. `saved` is false before the callback. As long as it's false, no text will be shown by `saveButtonText`. The text will show saving status after all